src/modules/form/components/group/TableGroup.tsx

```diff
@@ -0,0 +1,150 @@
+ *        "type": "GROUP", // represents first row
+ *        "item": [
+ *            // row 1 col 1 input
+ *            // row 1 col 2 input
+ *            // row 1 col 3 input
+ *        ]
+ *      },
+ *      {
+ *        "type": "GROUP", // represents second row
+ *        "item": [
+ *            // row 2 col 1 input
+ *            // row 2 col 2 input
+ *            // row 2 col 3 input
+ *        ]
+ *      }
+ *    ]
+ * }
+ */
+const TableGroup = ({
+  item,
+  renderChildItem,
+  viewOnly = false,
+}: GroupItemComponentProps) => {
+  const label = viewOnly ? item.readonlyText || item.text : item.text;
+
+  // Determine header row details by looking at the first row of Items
+  const tableHeaderInfo = useMemo(() => {
+    if (
+      !item.item ||
+      item.item[0].type !== ItemType.Group ||
+      !item.item[0].item
+    ) {
+      throw new Error(`invalid table format form item ${item.linkId}`);
+    }
+
+    return item.item[0].item.map(
+      ({ linkId, text, helperText, readonlyText, required }) => ({
+        id: `${linkId}-label`,
+        label: viewOnly ? readonlyText || text : text,
+        helperText,
+        required,
+      })
+    );
+  }, [item, viewOnly]);
+
+  return (
+    <Box>
+      {(label || item.helperText) && (
+        <Stack sx={{ mb: 2 }} gap={1}>
+          {label && <Typography>{label}</Typography>}
+          {item.helperText && (
+            <Typography variant='body2' component='span'>
+              {item.helperText}
+            </Typography>
+          )}
+        </Stack>
+      )}
+      <Table
+        sx={{
+          border: (theme) => `1px solid ${theme.palette.grey[200]}`,
+          p: 1,
+          '.MuiTableCell-root': {
+            borderRightColor: 'borders.light',
+            borderRightWidth: 1,
+            borderRightStyle: 'solid',
+          },
+        }}
+        aria-label={label || undefined}
+      >
+        <TableHead>
+          <TableRow>
+            {tableHeaderInfo.map(({ id, label, required, helperText }) => (
+              <TableCell
+                id={id}
+                sx={{
+                  borderBottomColor: 'borders.dark',
+                  borderBottomWidth: 2,
+                  borderBottomStyle: 'solid',
+                  py: 1,
+                }}
+              >
+                {label && (
+                  <RequiredLabel
+                    text={label}
+                    TypographyProps={{ fontWeight: 'bold' }}
+                    required={required}
+                  />
+                )}
+                {helperText && (
+                  <Typography
+                    variant='caption'
+                    component='span'
+                    display='block'
+                  >
+                    {helperText}
+                  </Typography>
+                )}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        {item.item?.map((rowItem) => (
+          <TableRow>
```

Comment:
Maybe use key props on rows/cols? Perhaps it doesn't matter since the table is static

Reply:
added 👍